pages/docs/tutorials/native/apple-framework.md

```diff
@@ -138,6 +142,44 @@ Most of the collection types are mapped to similar collection types from the oth
 {:.wide.zebra}
 
 
+### Kotlin Numbers and NSNumber
+
+The next part of the `<Framework>/Headers/Demo.h` contains number type mappings
+between Kotlin/Native and `NSNumber`. We have the base class called `DemoNumber` in Objective-C
+and `KotlinNumber` in Swift. It extends `NSNumber`.
+There are also child classes per Kotlin number type:
+
+|Kotlin|Swift|Objective-C| Simple type |
+-------|-----|-----------|
+|`-`      |`KotlinNumber` |`<Package>Number` | `-` |
+|`Byte`   |`KotlinByte`   |`<Package>Byte`   | `char` |
+|`UByte`  |`KotlinUByte`  |`<Package>UByte`  | `unsigned char` |
+|`Short`  |`KotlinShort`  |`<Package>Short`  | `short` |
+|`UShort` |`KotlinUShort` |`<Package>UShort` | `unsigned short` |
+|`Int`    |`KotlinInt`    |`<Package>Int`    | `int` |
+|`UInt`   |`KotlinUInt`   |`<Package>UInt`   | `unsigned int` |
+|`Long`   |`KotlinLong`   |`<Package>Long`   | `long` |
+|`ULong`  |`KotlinULong`  |`<Package>ULong`  | `unsigned long` |
+|`Float`  |`KotlinFloat`  |`<Package>Float`  | `float` |
+|`Double` |`KotlinDouble` |`<Package>Double` | `double` |
+|`Boolean`|`KotlinBoolean`|`<Package>Boolean`| `BOOL/Bool` |
+{:.wide.zebra}
+
+Every number type has a class method to create a new instance from a simple type. Also, there are all instance methods
+to extract a simple value back. Schematically, declarations look like that,
+
+<div clacss="sample" markdown="1" mode="obj-c" theme="idea" data-highlight-only auto-indent="false">
+
+```obj-c
+- (instancetype)initWith__TYPE_:(__TYPE__)value __attribute__((unavailable));
++ (instancetype)numberWith__TYPE_:(__TYPE__)value __attribute__((unavailable));
```

Comment:
` __attribute__((unavailable))` is used to mask constructors inherited from `NSNumber`, so the only allowed constructors are those without this attribute.

Reply:
fixed